midterm_project/shopping_list_archive.py

```diff
@@ -0,0 +1,125 @@
+import random
+
+annotations = ["-", "*", "+", "-"]
+annotate = random.choice(annotations) * 25
+
+shopping_list_archive = list()
+
+
+class PrintMixin:
+
+    def display_print(self):
+        print("{}\n {}\n{}".format(annotate, self._name, annotate))
+
+        for index, values in enumerate(self._ingredients, start=1):
+            print("{}. {}: {}".format(index, values.title(), self._ingredients[values]))
+
+        print('{}'.format(annotate))
+
+
+class Recipe(PrintMixin):
+
+    def __init__(self, name, ingredients):
+        self._name = name
+        self._ingredients = ingredients
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def ingredients(self):
+        return self._ingredients
+
+    def keys(self):
+        return self._ingredients.keys()
+
+    def __iter__(self):
+        return self._ingredients
+
+    def __len__(self):
+        return len(self._ingredients)
+
+    def __contains__(self, item):
+        return item in self._ingredients
+
+    def __repr__(self):
+        return self._name
+
+
+class RecipeBox:
+
+    def __init__(self):
+        self.recipes = []
+
+    # using the pick method we can
+    # extract a recipe by name or get
+    # a random recipe from the recipe box
+
+    def pick(self, name):
+        if name in self.recipes:
+            return name
+        else:
+            return random.choice(self.recipes)
+
+    def add(self, recipe):
+        self.recipes.append(recipe)
+
+    def delete(self, name):
+        try:
+            self.recipes.remove(name)
+            # return self.recipes
+        except ValueError as err:
+            print(f"Oops!! {err} occurred. Item no longer in the Recipe Box")
+
+    def list_recipes(self):
+        for recipe in self.recipes:
+            print(recipe.name, "\n")
+
+
+class Fridge(PrintMixin):
+
+    def __init__(self,ingredients={}):
+        self.ingredients = ingredients
+
+    def check_recipe(self,recipe):
+        pass
+
+    def add_item(self,ingredinent,quantity):
+        self.ingredients.update(ingredinent)
+        self.quantity.update(quantity)
```

Comment:
Is `self.ingredients` a dict with keys as ingredients and values as quantity? if so what you can do here is 
```
if self.ingredients.get(ingredient):
    self.ingredients[ingridient] += quantity
else
    self.ingredients[ingridient] = quantity
```
`.get()` will return None if the ingredient is not in the attribute ingredients, so what will happen is that the ingredient will be added to the dictionary if it is not already added and if it is the quantity will be updated by ``+=`